config/v1/types_infrastructure.go

```diff
@@ -691,7 +691,32 @@ type BareMetalPlatformLoadBalancer struct {
 
 // BareMetalPlatformSpec holds the desired state of the BareMetal infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type BareMetalPlatformSpec struct{}
+type BareMetalPlatformSpec struct {
+	// apiServerInternalIPs are the IP addresses to contact the Kubernetes API
+	// server that can be used by components inside the cluster, like kubelets
+	// using the infrastructure rather than Kubernetes networking. These are the
+	// IPs for a self-hosted load balancer in front of the API servers. In dual
+	// stack clusters this list contains two IPs otherwise only one.
+	//
+	// +kubebuilder:validation:Format=ip
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=set
+	APIServerInternalIPs []string `json:"apiServerInternalIPs"`
+
+	// ingressIPs are the external IPs which route to the default ingress
+	// controller. The IPs are suitable targets of a wildcard DNS record used to
+	// resolve default route host names. In dual stack clusters this list
+	// contains two IPs otherwise only one.
+	//
+	// +kubebuilder:validation:Format=ip
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=set
+	IngressIPs []string `json:"ingressIPs"`
+
+	// IP networks used to connect all the OpenShift cluster nodes.
```

Comment:
This godoc needs an update, what is the format of the MachineNetworks string?

Reply:
Updated by adding `Each network is provided in the CIDR format, e.g. "192.0.2.0/24"` in the description. I had a look at `types_network.go` and in there it's not at all handled so I think the description here is already better

```
	// allowedCIDRs is the list of allowed CIDRs.
	AllowedCIDRs []string `json:"allowedCIDRs,omitempty"`
```

I am not sure if we could do any further validation of the format given we need to support IPv4 and IPv6 (I am looking at how complex regex that would produce)